README.md

```diff
@@ -50,6 +50,39 @@ For more advanced configuration, please refer to the `curl` documentation.
 
 ## Development
 
+### System Requirements
+Install the following with your system's package manager
+ - opam
+ - make
+ - clang
+ - patch
+ - curl
+ - libcurl-devel
+
+Helper for opam install on linux:
+`bash -c "sh <(curl -fsSL https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh)"`
+
+
+### Setting up OPAM
+```bash
+opam init -y
+eval $(opam env --switch=default)
```

Comment:
I suggest avoiding the `-y` option for `opam init` since not everyone may want the default options.

Also, if you build other OCaml projects, you may want to use a switch specific to this project, to avoid dependency conflicts. Typically this is achieved with a local switch:

```bash
opam switch create . 4.14.1  # for OCaml 4.14.1
eval $(opam env)
```